praw/__init__.py

```diff
@@ -1294,6 +1298,60 @@ def is_oauth_session(self):
         """Return True when the current session is an OAuth2 session."""
         return isinstance(self._authentication, set)
 
+    def leave_contributor(self, subreddit=None, fullname=None):
+        """Abdicate approved submitter status in a subreddit. Use with care.
+
+        :param subreddit: The name of the subreddit to leave `status` from.
+        :param fullname: The fullname of the subreddit to leave `status` from.
+            `subreddit` and `fullname` cannot both be provided!
+
+        :returns: the json response from the server.
+        """
+        return self._leave_status(subreddit, fullname, status='contributor')
+
+    def leave_moderator(self, subreddit=None, fullname=None):
+        """Abdicate moderator status in a subreddit. Use with care.
+
+        :param subreddit: The name of the subreddit to leave `status` from.
+        :param fullname: The fullname of the subreddit to leave `status` from.
+            `subreddit` and `fullname` cannot both be provided!
+
+        :returns: the json response from the server.
+        """
+        self.evict(self.config['my_mod_subreddits'])
+        return self._leave_status(subreddit, fullname, status='moderator')
+
+    @decorators.restrict_access(scope='modself', mod=False)
+    def _leave_status(self, subreddit=None, fullname=None, status=None):
+        """Abdicate status in a subreddit.
+
+        :param subreddit: The name of the subreddit to leave `status` from.
+        :param fullname: The fullname of the subreddit to leave `status` from.
+            `subreddit` and `fullname` cannot both be provided!
+        :param status: The status to leave. One of 'moderator', 'contributor'.
+            Defaults to None as a precaution against accidents.
+            Please use :meth:`leave_contributor` or :meth:`leave_moderator`
+            rather than setting this directly.
+
+        :returns: the json response from the server.
+        """
+        if status == 'contributor':
```

Comment:
Perhaps you could pass in the URL, rather than a status since the status is only used to determine the url.